pkg/protocol/engine/congestioncontrol/rmc/rmc.go

```diff
@@ -0,0 +1,151 @@
+package rmc
+
+import (
+	"github.com/iotaledger/hive.go/ds/shrinkingmap"
+	"github.com/iotaledger/hive.go/ierrors"
+	"github.com/iotaledger/hive.go/runtime/syncutils"
+	"github.com/iotaledger/iota-core/pkg/model"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine/blocks"
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+type Manager struct {
+	// accumulated work from accepted blocks per slot
+	slotWork *shrinkingmap.ShrinkingMap[iotago.SlotIndex, iotago.WorkScore]
+
+	//reference mana cost per slot
+	rmc *shrinkingmap.ShrinkingMap[iotago.SlotIndex, iotago.Mana]
+
+	// eviction parameters
+	latestCommittedSlot   iotago.SlotIndex
+	commitmentEvictionAge iotago.SlotIndex
+
+	// commitment loader
+	commitmentLoader func(iotago.SlotIndex) (*model.Commitment, error)
+
+	// RMC parameters
+	rmcMin            iotago.Mana
+	rmcIncrease       iotago.Mana
+	rmcDecrease       iotago.Mana
+	increaseThreshold iotago.WorkScore
+	decreaseThreshold iotago.WorkScore
+
+	mutex syncutils.RWMutex
+}
+
+func NewManager(commitmentLoader func(iotago.SlotIndex) (*model.Commitment, error)) *Manager {
+	return &Manager{
+		slotWork:         shrinkingmap.New[iotago.SlotIndex, iotago.WorkScore](),
+		rmc:              shrinkingmap.New[iotago.SlotIndex, iotago.Mana](),
+		commitmentLoader: commitmentLoader,
+	}
+}
+
+func (m *Manager) SetRMCParameters(rmcMin iotago.Mana, rmcIncrease iotago.Mana, rmcDecrease iotago.Mana, increaseThreshold iotago.WorkScore, decreaseThreshold iotago.WorkScore) {
```

Comment:
You should use an APIProvider and get the correct RMC values for the current slot/epoch else this won't work with a protocol upgrade if the values change